backend/src/main/java/com/htecgroup/skynest/service/impl/BucketServiceImpl.java

```diff
@@ -114,6 +117,15 @@ public List<BucketResponse> listAllBuckets() {
         .collect(Collectors.toList());
   }
 
+  private boolean doesCurrentUserHaveViewPermissionOnBucket(BucketEntity bucket) {
+    try {
+      permissionService.currentUserHasPermissionForBucket(bucket.getId(), AccessType.VIEW);
```

Comment:
this method only checks the bucket for permissions

in order for this to work, when we grant permissions for a file or folder, we need to grant view permission for everything on the file/folder's path, bucket included

or rather, for everything on the path where the user doesn't already have some kind of permission

Reply:
Yeah, that was the plan